plugins/module_utils/utils.py

```diff
@@ -438,3 +441,35 @@ def wait_for_redfish_job_complete(redfish_obj, job_uri, job_wait=True, wait_time
         job_resp = redfish_obj.invoke_request("GET", job_uri, api_timeout=120)
         return job_resp, ""
     return job_resp, job_msg
+
+
+def get_dynamic_uri(idrac_obj, base_uri, search_label=''):
+    resp = idrac_obj.invoke_request(method='GET', uri=base_uri).json_data
+    if search_label and search_label in resp:
+        return resp[search_label]
+    return resp
+
+
+def get_scheduled_job_resp(idrac_obj, job_type):
+    # job_type can be like 'NICConfiguration' or 'BIOSConfiguration'
```

Comment:
We should add this in try block to catch exception as well.

Reply:
I think it's better to have exception handling in main function, rather than at each function, that makes no sense.